From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Taiyou06 <kaandindar21@gmail.com>
Date: Sun, 16 Feb 2025 01:13:04 +0100
Subject: [PATCH] Leaf: Directly use the pre-filtered ticking chunks list as
 the output

This patch uses already pre filtered chunks, which completely skips the isChunkNearPlayer check

diff --git a/net/minecraft/server/level/ServerChunkCache.java b/net/minecraft/server/level/ServerChunkCache.java
index 1d5a09c07b46b36ddcbe7c227ec368c4627f611c..eddee960ed5099b51a64c56c2dd8eca853c18879 100644
--- a/net/minecraft/server/level/ServerChunkCache.java
+++ b/net/minecraft/server/level/ServerChunkCache.java
@@ -573,14 +573,7 @@ public class ServerChunkCache extends ChunkSource implements ca.spottedleaf.moon
         if (output instanceof it.unimi.dsi.fastutil.objects.ObjectArrayList<LevelChunk> arrayList) arrayList.ensureCapacity(size); // Leaf - Use ensureCapacity to pre-populate the size of ticking chunks list output
 
         for (int i = 0; i < size; ++i) {
-            final ServerChunkCache.ChunkAndHolder chunkAndHolder = raw[i];
-            final LevelChunk levelChunk = chunkAndHolder.chunk();
-
-            if (!this.isChunkNearPlayer(chunkMap, levelChunk.getPos(), levelChunk)) {
-                continue;
-            }
-
-            output.add(levelChunk);
+            output.add(raw[i].chunk()); // Leaf - Directly use the pre-filtered ticking chunks list as the output
         }
         // Paper end - chunk tick iteration optimisation
     }
